app/src/main/java/nerd/tuxmobil/fahrplan/congress/settings/SettingsViewModel.kt

```diff
@@ -22,16 +30,29 @@ internal class SettingsViewModel(
             initialValue = SettingsUiState()
         )
 
+    private val effectsChannel = Channel<SettingsEffect>()
+    val effects = effectsChannel.receiveAsFlow()
+
+    private val activityResultKeys = mutableSetOf<String>()
+
     fun onViewEvent(event: SettingsEvent) {
-        handleEvent(event)
+        viewModelScope.launch {
+            handleEvent(event)
+        }
     }
 
-    private fun handleEvent(event: SettingsEvent) = when (event) {
+    private suspend fun handleEvent(event: SettingsEvent) = when (event) {
         DeviceTimezoneClicked -> toggleUseDeviceTimeZoneEnabled()
     }
 
-    private fun toggleUseDeviceTimeZoneEnabled() {
+    private suspend fun toggleUseDeviceTimeZoneEnabled() {
         val enabled = uiState.value.settings.isUseDeviceTimeZoneEnabled
         settingsRepository.setUseDeviceTimeZone(!enabled)
+        updateActivityResult(USE_DEVICE_TIME_ZONE_UPDATED)
+    }
+
+    private suspend fun updateActivityResult(key: String) {
+        activityResultKeys.add(key)
+        effectsChannel.send(SetActivityResult(activityResultKeys.toImmutableList()))
```

Comment:
Does it make sense to move the `viewModelScope.launch {` down here and remove the added `suspend` markers?